### PR TITLE
Expand tests for getValueByDataKey function

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Description

There were already tests present so I expanded on some edge cases.

## Related Issue

https://github.com/recharts/recharts/discussions/3717

## Motivation and Context

Testsss

## How Has This Been Tested?

Tests!

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Tesssstsss

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes
- [X] All new and existing tests passed.

### CODE REVIEW

File: test/util/ChartUtils.spec.tsx [hunk 1]
```diff
@@ -242,16 +242,94 @@ describe('parseScale', () => {
 });
 
 describe('getValueByDataKey', () => {
-  const data = { a: 1, b: 2, c: 3 };
+  const data: Record<string, unknown> = {
```
Comment (nikolasrieble) [inline test/util/ChartUtils.spec.tsx:245]: Nice, I think this type could be a nice one to generally use for a row of data.

Comment (PavelVanecek) [reply test/util/ChartUtils.spec.tsx:245]: Eventually I would like to find a solution that relies on generics instead of unknowns - but for now this will do

Comment (PavelVanecek) [reply test/util/ChartUtils.spec.tsx:245]: Also I should really be using `PropertyKey` instead, and adding a test of what happens when DataKey returns a symbol!

Comment (nikolasrieble) [reply test/util/ChartUtils.spec.tsx:245]: A side note: I used Symbols in a few workaround to add an index to the data before passing it to recharts. I then use the index to identify an active datapoint from a mouseOver, or a Tooltip. 

So for my use case this would have to be 

```
Record<string | Symbol, unknown>
```

Comment (nikolasrieble) [reply test/util/ChartUtils.spec.tsx:245]: Unrelated to this PR, but generally relevant: 

Should we possibly do this in the library itself? The whole getActiveIndex is quite expensive (iterating over the whole dataset every time) and could be done much simpler if we just add an index with a Symbol to the dataset as a preprocessing step in the Chart.

**Review by nikolasrieble — APPROVED** [review]
❤️
